File: app/src/main/java/com/lambdaschool/congressdata/CongresspersonProfileViewModel.java
package com.lambdaschool.congressdata;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.ViewModel;

public class CongresspersonProfileViewModel extends ViewModel {
    private LiveData<CongresspersonProfile> profile;
    private String                          id;

    public void init(String id) {
        this.id = id;
    }

    public LiveData<CongresspersonProfile> getProfile() {
        if (profile == null) {
            loadProfile();
        }
        return profile;
    }

    private void loadProfile() {
        profile = ProfileRepository.getProfile(id);
    }
}

File: app/src/main/java/com/lambdaschool/congressdata/CongresspersonProfileViewModel.kt
package com.lambdaschool.congressdata

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.ViewModel

class CongresspersonProfileViewModel(var id: String?) : ViewModel() {
    var profile: LiveData<CongresspersonProfile>? = null
    get() {
        return if (field == null) {
            loadProfile()
        } else {
            field
        }
    }

    private fun loadProfile(): LiveData<CongresspersonProfile>? {
        profile = ProfileRepository.getProfile(id)
        return profile
    }
}
